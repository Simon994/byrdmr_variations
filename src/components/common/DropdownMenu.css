/* Drop-down menu (for tablet width and below) */

.outer-menu-container {
  display: flex;
  justify-content: center;
}

.outer-menu-container .byrd-logo {
  margin-top: 52px;
}

.inner-menu-container {
  position: absolute;
  right: 63px;
  display: flex;
  justify-content: center;
  align-items: center;
}

.menu-trigger {
  background: none;
  cursor: pointer;
  border: none;
  transition: box-shadow 0.4s ease;
  margin-left: auto;
}

.menu-trigger:hover {
  box-shadow: 0 1px 8px rgba(0, 0, 0, 0.3);
}

.menu-trigger span {
  font-weight: 700;
  vertical-align: middle;
  font-size: 14px;
  margin: 0 10px;
}

.menu-trigger img {
  margin-top: 40px;
  width: 60px;
}

.menu {
  background: white;
  position: absolute;
  top: 60px;
  right: 0;
  width: 150px;
  box-shadow: 0 1px 8px rgba(0, 0, 0, 0.3);
  opacity: 0;
  visibility: hidden;
  transform: translateY(-20px);
  transition: opacity 0.4s ease, transform 0.4s ease, visibility 0.4s;
}

.menu.active {
  opacity: 1;
  visibility: visible;
  transform: translateY(50px);
}

.menu ul {
  list-style: none;
  padding: 0;
  margin: 0;
}

.menu li {
  border-bottom: 1px solid #ddd;
}

@media screen and (max-width: 375px) {
  .outer-menu-container .byrd-logo {
    margin-top: 25px;
    width: 136px;
    height: auto;
  }

  .menu-trigger img {
    margin-top: 7px;
    position: absolute;
    right: -45px;
  }

  .menu {
    z-index: 20;
    top: 30px;
  }

  form {
    width: 322px;
  }
}
