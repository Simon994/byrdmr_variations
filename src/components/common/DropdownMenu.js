import {useRef} from 'react'

import './DropdownMenu.css'
import MenuAndNavItems from './MenuAndNavItems'
import ToggleActive from './ToggleActive'
import menu from '../../styles/assets/menu.png'
import Logo from '../../styles/assets/Logo.png'

const DropdownMenu = () => {
  const dropdownRef = useRef(null)
  const [isActive, setIsActive] = ToggleActive(dropdownRef, false)
  const onClick = () => setIsActive(!isActive)

  return (
    <div className='outer-menu-container'>
      <img 
        src={Logo}
        alt='ByrdMR logo'
        className='byrd-logo'
      />
      <div className='inner-menu-container'>
        <button onClick={onClick} className='menu-trigger'>
          <img
            src={menu}
            alt='Menu button icon'
          />
        </button>
        <nav
          ref={dropdownRef}
          className={`menu ${isActive ? 'active' : 'inactive'}`}
        >
          <MenuAndNavItems 
            listItems={['store', 'tutorials']}/>
          <MenuAndNavItems 
            listItems={['blog', 'contact']}/>
        </nav>
      </div>
    </div>
  )
}

export default DropdownMenu