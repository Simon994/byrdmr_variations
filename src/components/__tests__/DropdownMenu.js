import React from 'react'
import { BrowserRouter } from 'react-router-dom'
import renderer from 'react-test-renderer'
import DropdownMenu from '../common/DropdownMenu'

describe('The DropdownMenu', () => {
  it('renders as expected', () => {
    const tree = renderer.create(
      <BrowserRouter>
        <DropdownMenu/>
      </BrowserRouter>
    ).toJSON()

    expect(tree).toMatchSnapshot()
  })
})
