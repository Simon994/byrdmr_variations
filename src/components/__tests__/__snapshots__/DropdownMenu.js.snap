// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`The DropdownMenu renders as expected 1`] = `
<div
  className="inner-menu-container"
>
  <button
    className="menu-trigger"
    onClick={[Function]}
  >
    <img
      alt="Menu button icon"
      src="menu.png"
    />
  </button>
  <nav
    className="menu inactive"
  >
    <ul>
      <li>
        <a
          aria-current={null}
          className="nav-link"
          href="/store"
          onClick={[Function]}
        >
          STORE
        </a>
      </li>
      <li>
        <a
          aria-current={null}
          className="nav-link"
          href="/tutorials"
          onClick={[Function]}
        >
          TUTORIALS
        </a>
      </li>
    </ul>
    <ul>
      <li>
        <a
          aria-current={null}
          className="nav-link"
          href="/blog"
          onClick={[Function]}
        >
          BLOG
        </a>
      </li>
      <li>
        <a
          aria-current={null}
          className="nav-link"
          href="/contact"
          onClick={[Function]}
        >
          CONTACT
        </a>
      </li>
    </ul>
  </nav>
</div>
`;
